### DISCUSSION TITLE
Using modules from multiple environments?

### DISCUSSION THREAD

**Discussion by benjamin-cash** [JCSDA/spack-stack#1131] [Q&A]
I've installed modules from a few different environment templates on Frontera, and it seems as though the way things are set up you can only have one set of modules active at a time. Is this by design? For example, gsi-addon appears to be lacking many of the modules available via ufs-weather-model, but loading one will cause the other to be automatically unloaded.

**Comment by climbfuji** [comment]
Yes, that is by design. Did you run `spack module lmod refresh --upstream-modules` for the chained environment? If yes, and there are still modules missing, then these need to be added to gsi-addon-env.

**Reply by benjamin-cash** [reply]
Aha! No, I didn't run the --upstream-modules, that's the step I was missing. Thanks!
